marlowe/src/Language/Marlowe/Analysis/Safety/Types.hs

```diff
@@ -91,7 +109,254 @@ data SafetyError =
   | WrongNetwork
     -- | The contract contains an illegal ledger address.
   | IllegalAddress Ledger.Address
-    deriving (Eq, Show)
+    -- | The safety analysis exceeded the allotted time.
+  | SafetyAnalysisTimeout
+    deriving (Eq, Generic, Show)
+
+
+instance ToJSON SafetyError where
+  toJSON MissingRolesCurrency =
+    object
+      [ "error" .= ("MissingRolesCurrency" :: String)
```

Comment:
It's surprising that you need to help the type inference to serialize here (`xxxx:: String`)

Reply:
Not at all, with `OverloadedStrings` the type of a string literal is polymorphic, because it is desugarred as

```hs
fromString ("MissingRolesCurrency" :: String)
```

And in this context, we have

```hs
(.= "error") :: forall a. ToJSON a => a ->Pair
```

So the type `a` is polymorphic in both its creation (the output of `fromString`) and elimination (the input of `(.= "error")`), thus ambiguous.